File: backend/src/main/java/com/soen345/ticketreservation/model/Event.java
package com.soen345.ticketreservation.model;

import com.soen345.ticketreservation.model.enums.EventCategory;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "events")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Title is required")
    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @NotNull(message = "Date is required")
    @Column(nullable = false)
    private LocalDateTime date;

    @NotBlank(message = "Location is required")
    @Column(nullable = false)
    private String location;

    @NotNull(message = "Category is required")
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EventCategory category;

    @Min(value = 1, message = "Capacity must be at least 1")
    @Column(nullable = false)
    private int capacity;

    @NotNull(message = "Price is required")
    @Column(nullable = false)
    private BigDecimal price;

    @Version
    private Long version;
}

File: app/src/main/java/com/example/frontend/model/Event.kt
package com.example.frontend.model

import java.math.BigDecimal

enum class EventCategory {
    MOVIE, CONCERT, TRAVEL, SPORTS
}

data class EventResponse(
    val id: Long,
    val title: String,
    val description: String?,
    val date: String,
    val location: String,
    val category: EventCategory,
    val capacity: Int,
    val price: BigDecimal,
    val availableSeats: Int
)

data class CreateEventRequest(
    val title: String,
    val description: String?,
    val date: String,
    val location: String,
    val category: EventCategory,
    val capacity: Int,
    val price: BigDecimal
)
